browser/lib/util/bufferutils.js

```diff
@@ -90,19 +90,19 @@ var BufferUtils = (function() {
 	BufferUtils.base64Decode = function(str) {
 		if(ArrayBuffer)
 			return base64ToArrayBuffer(str);
-
-		if(CryptoJS)
-			return CryptoJS.enc.Base64.parse(str);
+		return CryptoJS.enc.Base64.parse(str);
 	};
 
 	BufferUtils.utf8Encode = function(string) {
-		if(CryptoJS)
-			return CryptoJS.enc.Utf8.parse(string);
+		return CryptoJS.enc.Utf8.parse(string);
 	};
 
 	BufferUtils.utf8Decode = function(buf) {
-		if(CryptoJS)
+		if(isArrayBuffer(buf))
+			buf = BufferUtils.toWordArray(buf) // CryptoJS only works with WordArrays
+		if(isWordArray(buf))
 			return CryptoJS.enc.Utf8.stringify(buf);
+		throw new Error("Expected input of utf8Decode to be a buffer");
```

Comment:
They see `Ably: RealtimeChannel.onMessage(): Unexpected error decoding message; err = [Error: Expected input of utf8Decode to be a buffer]`, or with errorInfo:  `Ably: RealtimeChannel.onMessage(): Unexpected error decoding message; err = [ErrorInfo: Expected input of utf8Decode to be a buffer; statusCode=123; code=12345]` (or whatever)

(so the statusCode is not present in the final emitted error either way - it's emitted by https://github.com/ably/ably-js/blob/master/common/lib/client/realtimechannel.js#L267)

Can change the error message to `'Message had encoding component utf-8, which ably-js expect to be represented as a binary data buffer or CryptoJS WordArray; but the given data was a ' + typeof buf` or something. What statusCode were you thinking? None seem to really fit afaics.